pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go

```diff
@@ -1,20 +1,81 @@
 package certgraphanalysis
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+	corev1 "k8s.io/api/core/v1"
 )
 
-type caBundleRewriteFunc func(caBundle *certgraphapi.CertificateAuthorityBundle)
+const lastRewrittenByAnnotation = "openshift.io/last-rewritten-by"
 
+type configMapRewriteFunc func(configMap *corev1.ConfigMap)
+type secretRewriteFunc func(secret *corev1.Secret)
+type caBundleRewriteFunc func(caBundle *certgraphapi.CertificateAuthorityBundle)
 type certKeyPairRewriteFunc func(certKeyPair *certgraphapi.CertKeyPair)
 
 type metadataOptions struct {
 	rewriteCABundle    caBundleRewriteFunc
 	rewriteCertKeyPair certKeyPairRewriteFunc
+	rewriteConfigMap   configMapRewriteFunc
```

Comment:
I think we'll want to split these into a different kind of option in the future, but not require now.